### DISCUSSION TITLE
ZoneTree vs FASTERKV

### DISCUSSION THREAD

**Discussion by Xhanti** [koculu/ZoneTree#102] [Q&A] [answered]
Hi,

This project looks very interesting to me but I'm wondering if it's any different from FASTERKV? From my reading it looks like you're trying to solve similar problems in a bit of a different way. I'm wondering which library to rely on for a future project (simple CDN).

Regards

**Comment by koculu** [comment, marked as answer]
Hi,
FasterKV is a HashTable, and ZoneTree is a log structured merge tree. ([LSM-Tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree))
In other words, FasterKV is a Key-Value-Store. ([KVS](https://en.wikipedia.org/wiki/Key%E2%80%93value_database))
ZoneTree is an Ordered-Key-Value-Store. ([OKVS](https://en.wikipedia.org/wiki/Ordered_Key-Value_Store))

The keys in ZoneTree are ordered, that is the key difference.

**Reply by Xhanti** [reply]
Hi, 

Thank you for clearing that up.

Regards
